Transactions.md

```diff
@@ -1,182 +1,292 @@
+	- witness <----	"witness" is used instead of "script_sig" for
+ 			transaction validation; however, "witness" is stored
+ 			separately, and the cost for its size is smaller. So,
+ 		    the calculation of ordinary data is separated
+ 			from the witness data.
+	- sequence: 4 bytes
+
+witness_header: 2 bytes
+	- flag: 1 byte
+	- marker: 1 byte
+
+witness: 108 bytes
+	- number_of_witness_elements: 1 byte
+	- sig_length: 1 byte
+	- sig: 72 bytes
+	- pub_key_length: 1 byte
+	- pub_key: 33 bytes
+
+funding_transaction: 71 + offer_change_spk_script length + accept_change_spk_script length + 41 * num_inputs bytes
+	- version: 4 bytes
+	- witness_header <---- part of the witness data
+	- count_tx_in: 1 byte
+	- tx_in: 41 bytes * num_inputs
+		funding_input
+	- count_tx_out: 1 byte
+	- tx_out: 61 + offer_change_spk_script length + accept_change_spk_script length
+		funding_ouptut (43 bytes),
+		change_output (9 + offer_change_spk_script length bytes),
+		change_output (9 + accept_change_spk_script length bytes)
+	- lock_time: 4 bytes
+```
+
+Multiplying non-witness data by 4 results in a weight of:
+
+```
+// total_change_length = offer_change_spk_script length + accept_change_spk_script length
+// 284 + 4 * total_change_length + 164 * num_inputs weight
+funding_transaction_weight = 4 * funding_transaction
+
+// 2 + 108 * num_inputs weight
+witness_weight = witness_header + witness * num_inputs
+
+overall_funding_tx_weight = 286 + 4 * total_change_length + 272 * num_inputs weight
+```
+
+## Expected Weight of the Contract Execution or Refund Transaction
+
+The *expected weight* of a contract execution or refund transaction is calculated as follows:
+
+```
+multi_sig: 71 bytes
+	- OP_2: 1 byte
+	- OP_DATA: 1 byte (offer_pubkey length)
+	- offer_pubkey: 33 bytes
+	- OP_DATA: 1 byte (accept_pubkey length)
+	- accept_pubkey: 33 bytes
+	- OP_2: 1 byte
+	- OP_CHECKMULTISIG: 1 byte
+
+offer_output: 9 + offer_output_script length bytes
+	- value: 8 bytes
+	- var_int: 1 byte (pk_script length)
+	- pk_script: offer_output_script length
+
+accept_output: 9 + accept_output_script length bytes
+	- value: 8 bytes
+	- var_int: 1 byte (pk_script length)
+	- pk_script: accept_output_script length
+
+funding_tx_input: 41 bytes
+	- previous_out_point: 36 bytes
+		- hash: 32 bytes
+		- index: 4 bytes
+	- var_int: 1 byte (script_sig length)
+	- script_sig: 0 bytes
+	- witness <----	"witness" is used instead of "script_sig" for
+ 			transaction validation; however, "witness" is stored
+ 			separately, and the cost for its size is smaller. So,
+ 		    the calculation of ordinary data is separated
+ 			from the witness data.
+	- sequence: 4 bytes
+
+witness_header: 2 bytes
+	- flag: 1 byte
+	- marker: 1 byte
+
+witness: 222 bytes
+	- number_of_witness_elements: 1 byte
+	- nil_length: 1 byte
+	- sig_offer_length: 1 byte
+	- sig_offer: 73 bytes
+	- sig_accept_length: 1 byte
+	- sig_accept: 73 bytes
+	- witness_script_length: 1 byte
+	- witness_script (multi_sig): 71 bytes
```

Comment:
I think this may be in the wrong order? If this is the "top" of the stack, that means the first sig "popped" off is `sig_accept`, however in funding output specification you said the order has the `offer_pubkey` first?

>multi_sig: 71 bytes
	- OP_2: 1 byte
	- OP_DATA: 1 byte (offer_pubkey length)
	- offer_pubkey: 33 bytes
	- OP_DATA: 1 byte (accept_pubkey length)
	- accept_pubkey: 33 bytes
	- OP_2: 1 byte
	- OP_CHECKMULTISIG: 1 byte

Reply:
https://github.com/lightningnetwork/lightning-rfc/blob/master/03-transactions.md#expected-weight-of-the-commitment-transaction

Reply:
I'll assume they got it right then :+1: